core/rfb.js

```diff
@@ -1271,6 +1283,127 @@ export default class RFB extends EventTargetMixin {
         return this._fail("No supported sub-auth types!");
     }
 
+    _negotiate_ra2ne_auth() {
```

Comment:
This handling is very complex, so we need some unit tests for this to make sure it won't break in the future.